planner/core/plan_cache.go

```diff
@@ -208,55 +207,42 @@ func getPointQueryPlan(stmt *ast.Prepared, sessVars *variable.SessionVars, stmtC
 	return plan, names, true, nil
 }
 
-func getGeneralPlan(sctx sessionctx.Context, cacheKey kvcache.Key, bindSQL string,
-	is infoschema.InfoSchema, stmt *PlanCacheStmt, paramTypes []*types.FieldType) (Plan,
+func getGeneralPlan(sctx sessionctx.Context, cacheKey kvcache.Key,
+	bindSQL string, stmt *PlanCacheStmt, paramTypes []*types.FieldType) (Plan,
 	[]*types.FieldName, bool, error) {
 	sessVars := sctx.GetSessionVars()
 	stmtCtx := sessVars.StmtCtx
 
-	if cacheValue, exists := sctx.PreparedPlanCache().Get(cacheKey); exists {
-		if err := CheckPreparedPriv(sctx, stmt, is); err != nil {
-			return nil, nil, false, err
-		}
```

Comment:
It seems we didn't check this part in the new codebase.

Reply:
Nice catch.

Reply:
Updated, PTAL.